src/Microsoft.SqlTools.ServiceLayer/Admin/Database/DatabasePrototypeAzure.cs

```diff
@@ -91,58 +91,58 @@ public string AzureEditionDisplay
-            //             // Instead of creating db instance with default Edition, update EditionToCreate while selecting Edition from the UI.
-            //             this.EditionToCreate = MapAzureEditionToDbEngineEdition(edition);
-            //             string storageAccountType = AzureServiceLevelObjectiveProvider.TryGetAzureStorageType(value, AzureServiceLocation);
-            //             if (!string.IsNullOrEmpty(storageAccountType))
-            //             {
-            //                 this.currentState.backupStorageRedundancy = storageAccountType;
-            //             }
-
-            //             // Try to get the azure maxsize from the api,if not the default hardcoded maxsize will be retrieved.
-            //             DbSize dbSize = AzureServiceLevelObjectiveProvider.TryGetAzureMaxSize(value, serverLevelObjective, AzureServiceLocation);
-            //             if (!string.IsNullOrEmpty(dbSize.ToString()))
-            //             {
-            //                 this.currentState.maxSize = new DbSize(dbSize.Size, dbSize.SizeUnit);
-            //             }
-            //         }
-            //         else
-            //         {
-            //             if (edition == this.currentState.azureEdition)
-            //             { //No changes, return early since we don't need to do any of the changes below
-            //                 return;
-            //             }
-
-            //             this.currentState.azureEdition = edition;                        
-            //             this.EditionToCreate = MapAzureEditionToDbEngineEdition(edition);
-            //             this.CurrentServiceLevelObjective = AzureSqlDbHelper.GetDefaultServiceObjective(edition);
-            //             this.BackupStorageRedundancy = AzureSqlDbHelper.GetDefaultBackupStorageRedundancy(edition);
-            //             var defaultSize = AzureSqlDbHelper.GetDatabaseDefaultSize(edition);
-
-            //             this.MaxSize = defaultSize == null ? String.Empty : defaultSize.ToString();
-            //         }
-            //         this.NotifyObservers();
-            //     }
-            //     else
-            //     {
-            //         //Can't really do much if we fail to parse the display name so just leave it as is and log a message
-            //         System.Diagnostics.Debug.Assert(false,
-            //             string.Format(CultureInfo.InvariantCulture,
-            //                 "Failed to parse edition display name '{0}' back into AzureEdition", value));
-            //     }
-            // }
+            set
+            {
+                AzureEdition edition;
+                if (AzureSqlDbHelper.TryGetAzureEditionFromDisplayName(value, out edition))
+                {
+                    //Try to get the ServiceLevelObjective from the api,if not the default hardcoded service level objectives will be retrieved.
```

Comment:
This section deals with retrieving the SLO details dynamically. I'll look into enabling it after merging this PR.

Reply:
We don't have this provider file in STS yet, so I'd have to port it over from SSMS.